src/rest/domain/GenericDomainObject.js

```diff
@@ -0,0 +1,13 @@
+export default class GenericDomainObject {
+    constructor(client, props = {}) {
+        if (!client) {
+            throw new Error('A client must be given')
+        }
+        this._client = client
+        Object.assign(this, props)
+    }
+
+    setProps(props = {}) {
```

Comment:
Maybe `setState` for parity with React's Component API?

Reply:
That's a good point. In a way I'd like to use some completely different word than `props` or `state`, but definitely from those two `state` is better